### PR TITLE
[WIP] update's from David to get HTCONDOR batch jobs

### PR DESCRIPTION
**Pull request by davidjamin**

### CODE REVIEW

File: analyzers/ResonanceBuilder.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import pprint 
 import itertools
 
-mass = {23: 91, 24: 80.4, 25: 125}
+mass = {23: 91, 25: 125, 32: 1000}
```
Comment (cbernet) [inline analyzers/ResonanceBuilder.py]: you should not remove 24 (the W). 
What is 32?

File: analyzers/ResonanceBuilder.py [hunk 2]
```diff
@@ -49,5 +49,8 @@ def process(self, event):
             resonances.append( Resonance(leg1, leg2, self.cfg_ana.pdgid) )
         # sorting according to distance to nominal mass
         nominal_mass = mass[self.cfg_ana.pdgid]
-        resonances.sort(key=lambda x: abs(x.m()-nominal_mass))
+        if self.cfg_ana.pdgid==32:
```
Comment (cbernet) [inline analyzers/ResonanceBuilder.py]: This should not be necessary, there is no reason to treat a given type of resonance differently

File: framework/looper.py [hunk 3]
```diff
@@ -167,14 +168,14 @@ def doSigUsr2(sig,frame):
...
-        # save the versions
-        if self.config.versions:
+        
+        if hasattr(self.config, 'versions'):
             self.config.versions.write_yaml('/'.join([self.outDir,
                                                       'software.yaml']))        
         # remove versions from the config as it can't be pickled
-        config_no_versions = copy.copy(self.config)
-        delattr(config_no_versions, 'versions')
+        #config_no_versions = copy.copy(self.config)
```
Comment (cbernet) [inline framework/looper.py]: I don't see how this version stuff could be related to these condor developments -> should stay.
What is the problem with that?

File: utils/batchmanager.py [hunk 4]
```diff
@@ -44,8 +44,15 @@ def DefineOptions(self):
                                 dest="negate", default=False,
                                 help="create jobs, but does not submit the jobs.")
         self.parser_.add_option("-b", "--batch", dest="batch",
-                                help="batch command. default is: 'bsub -q 8nh < batchScript.sh'. You can also use 'nohup < ./batchScript.sh &' to run locally.",
-                                default="bsub -q 8nh < ./batchScript.sh")      
+                                help="""batch command to submit job. 
+                                    ==> OLD!!!! LSF submission to a queue, e.g. 8nh:
```
Comment (cbernet) [inline utils/batchmanager.py]: why keeping LSF stuff? it seems to me that the supporting shell script has been removed.
Was LSF fully decommissioned ? If yes, all LSF stuff should just go

File: utils/batchmanager.py [hunk 5]
```diff
@@ -44,8 +44,15 @@ def DefineOptions(self):
...
+                                          'bsub -q 8nh < ./batchScript.sh'. You can also use 'nohup < ./batchScript.sh &' to run locally.
+                                    ==> HTCondor submission using HTCondor internal files transfer
+                                    ==> with max 240 minutes of wall clock runtime:
+                                          './run_condor.sh -t 240' 
+                                    The default is '%default'.""",
+                                #default="bsub -q 8nh < ./batchScript.sh")
```
Comment (cbernet) [inline utils/batchmanager.py]: remove

File: utils/batchmanager.py [hunk 6]
```diff
@@ -240,28 +256,34 @@ def RunningMode(self, batch):
 
         '''Return "LXPUS", "PSI", "NAF", "LOCAL", or None,
 
-        "LXPLUS" : batch command is bsub, and logged on lxplus
-        "PSI"    : batch command is qsub, and logged to t3uiXX
-        "NAF"    : batch command is qsub, and logged on naf
-        "IC"     : batch command is qsub, and logged on hep.ph.ic.ac.uk
-        "LOCAL"  : batch command is nohup.
+        "LXPLUS(LSF)": batch command is bsub, and logged on lxplus
```
Comment (cbernet) [inline utils/batchmanager.py]: remove LSF stuff?

File: utils/batchmanager.py [hunk 7]
```diff
@@ -240,28 +256,34 @@ def RunningMode(self, batch):
...
         onLxplus = hostName.startswith('lxplus')
         onPSI    = hostName.startswith('t3ui')
-        onNAF =  hostName.startswith('naf')
+        onNAF    = hostName.startswith('naf')
 
         batchCmd = batch.split()[0]
         
         if batchCmd == 'bsub':
-            if not onLxplus:
+            print 'running on LSF : %s from %s' % (batchCmd, hostName)
```
Comment (cbernet) [inline utils/batchmanager.py]: remove

File: utils/batchmanager.py [hunk 8]
```diff
@@ -240,28 +256,34 @@ def RunningMode(self, batch):
...
+            err = "not workable any more -> NO JOBS SUBMITTED !!! use CONDOR"
+            raise ValueError( err )
+
+        elif batchCmd.find("run_condor.sh")>=0:
```
Comment (cbernet) [inline utils/batchmanager.py]: replace by a more pythonic statement like 

```python
if 'run_condor.sh' in batchCmd
```

File: utils/versions.py [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import modulefinder
-import git
+#import git
```
Comment (cbernet) [inline utils/versions.py]: must be kept

File: utils/versions.py [hunk 10]
```diff
@@ -24,9 +24,9 @@ def __init__(self, scriptfname):
 #        finder.report()
         self.tracked = dict()
         processed = []
-        for key, mod in finder.modules.iteritems():
-            #if mod.__path__ and os.environ['USER'] in mod.__path__[0]:
-            self._analyze(key, mod)
+        #for key, mod in finder.modules.iteritems():
```
Comment (cbernet) [inline utils/versions.py]: must be kept

**Comment by vvolkl** [comment]
Hi Colin,
yes, gitpython has been available for a while in the central fcc installation, it there is an import error, it must come from an old configuration.

**Comment by JavierCVilla** [comment]
> @vvolkl @JavierCVilla did you install gitpython by default in the last releases? (not sure which one David is using). It's really needed.

As far as I can see it is installed and working:

```
$ source /cvmfs/fcc.cern.ch/sw/views/releases/externals/latest/x86_64-slc6-gcc62-opt/setup.sh 
$ python
Python 2.7.15 (default, Jul 30 2018, 09:38:14) 
[GCC 6.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from git import Repo
>>> import git
>>> 
```

File: analyzers/ResonanceBuilder.py (hunk 1 above)
Comment (davidjamin) [reply analyzers/ResonanceBuilder.py]: 32 is Z'

File: analyzers/ResonanceBuilder.py (hunk 2 above)
Comment (davidjamin) [reply analyzers/ResonanceBuilder.py]: I ll check why we have done this

File: framework/looper.py (hunk 3 above)
Comment (davidjamin) [reply framework/looper.py]: need to clarify

File: utils/versions.py (hunk 9 above)
Comment (davidjamin) [reply utils/versions.py]: need to clarify

File: utils/versions.py (hunk 10 above)
Comment (davidjamin) [reply utils/versions.py]: to clarify

File: utils/versions.py (hunk 9 above)
Comment (davidjamin) [reply utils/versions.py]: OK it was a temporary issue from SW long time ago, need to get it back

Comment (cbernet) [reply utils/versions.py]: I think you can just find the commit(s) where these changes have been introduced in this PR and just revert them by pushing to this branch.

**Comment by vvolkl** [comment]
@cbernet I think this PR was still work in progress. David mentioned that there are still a lot of failing jobs with condor, so we are looking into that. You can leave this and we  can open  another one, but just keep in mind  condor submission is not production ready yet.

**Comment by cbernet** [comment]
@vvolkl I know that condor submission is not ready, and I understood that the issue needs to be fixed on the CERN IT side. @davidjamin told me that the PR had been updated and did not ask me to hold it. If a PR needs to be hold, please mention it in the comments or directly to me in private before pushing

**Comment by vvolkl** [comment]
It was marked with [WIP] (Work in Progress) in the title ;)

**Comment by cbernet** [comment]
I'd prefer to have plain PR messages and, when contacted on skype, to be explicitely told to hold the PR. 

I was just told that on skype that the PR had been updated and that we now had to wait for the fixes from the help desk.  

Please guys, I have very very little time for these things, I have left FCC, and I'm just looking at this to help you.

**Comment by vvolkl** [comment]
And that's highly appreciated! Sure, I just didn't get notified that the problem was now CERN IT side only, that's why I was still looking into the errrors. Sorry for the spam

**Pull request merged**